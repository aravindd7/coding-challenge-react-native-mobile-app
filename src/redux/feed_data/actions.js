export const FEED_DATA__UPDATE = "FEED_DATA__UPDATE";

export const feedData_Update = ({ data }) => ({
  type: FEED_DATA__UPDATE,
  payload: { data }
});
