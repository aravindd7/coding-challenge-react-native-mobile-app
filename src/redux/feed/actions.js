import { feedData_Update } from "../feed_data/actions";

export const FEED_UPDATE__REQUEST = "FEED_UPDATE__REQUEST";
export const FEED_UPDATE__FAIL = "FEED_UPDATE__FAIL";
export const FEED_UPDATE__SUCCESS = "FEED_UPDATE__SUCCESS";

export const FEED_SECTION_CHANGE = "FEED_SECTION_CHANGE";

export const changeSection = ({ section }) => {
  return (dispatch, getState) => {
    // TODO changeSection Action Creator
  };
};

export const feedUpdate = () => {
  return (dispatch, getState) => {
    // TODO feedUpdate Action Creator
  };
};

export const feedUpdate_Request = () => ({
  type: FEED_UPDATE__REQUEST,
  payload: {}
});

export const feedUpdate_Fail = ({ error }) => ({
  type: FEED_UPDATE__FAIL,
  payload: { error }
});

export const feedUpdate_Success = () => ({
  type: FEED_UPDATE__SUCCESS,
  payload: {}
});

export const feedSectionChange = ({ section }) => ({
  type: FEED_SECTION_CHANGE,
  payload: { section }
});
