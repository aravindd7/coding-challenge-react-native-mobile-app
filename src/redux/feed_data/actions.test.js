import * as actions from "./actions";

it("Creates Action feedData_Update", () => {
  const action = actions.feedData_Update({
    data: [
      { id: 1, url: "url-1", post: "post-1" },
      { id: 1, url: "url-1", post: "post-1" }
    ]
  });
  expect(action).toMatchSnapshot();
});
