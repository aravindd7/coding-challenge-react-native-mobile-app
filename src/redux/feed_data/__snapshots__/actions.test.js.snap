// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates Action feedData_Update 1`] = `
Object {
  "payload": Object {
    "data": Array [
      Object {
        "id": 1,
        "post": "post-1",
        "url": "url-1",
      },
      Object {
        "id": 1,
        "post": "post-1",
        "url": "url-1",
      },
    ],
  },
  "type": "FEED_DATA__UPDATE",
}
`;
